### PR TITLE
Feature/callbacks

### PR DESCRIPTION
**Pull request by alexeyklyukin**
Move the functionality that was provided by the AWS module (tagging the EC2 instances/EBS) into the callbacks, that can perform arbitrary actions on various events in Patroni, i.e. start, stop, restart, but also promote/demote.

Move the aws.py from helpers/ to scripts/, a new location for pluggable functionality.

Update unit-tests accordingly.

### CODE REVIEW

File: helpers/postgresql.py [hunk 1]
```diff
@@ -240,6 +242,26 @@ def is_leader(self):
...
+        if not self.callback or cb_name not in self.callback:
+            return False
+        cmd = self.callback[cb_name]
+        if is_leader is None:
+            try:
+                is_leader = self.is_leader(check_only=True)
+            except psycopg2.OperationalError as e:
+                logger.warning("unable to perform {0} action, cannot obtain the cluster role: {1}".format(cb_name, e))
+                return False
+        name = self.scope
```
Comment (sli67) [inline helpers/postgresql.py]: Wouldn't "scope" be a better variable name for self.scope?

File: helpers/postgresql.py [hunk 2]
```diff
@@ -253,18 +275,37 @@ def start(self):
         ret = subprocess.call(self._pg_ctl + ['start', '-o', self.server_options()]) == 0
         ret and self.load_replication_slots()
         self.save_configuration_files()
-        if self.on_change_callback:
-            self.on_change_callback('replica' if os.path.exists(self.recovery_conf) else 'master')
+        if ret and 'on_start' in self.callback:
```
Comment (sli67) [inline helpers/postgresql.py]: Shouldn't such magic strings better be declared as constants?

Comment (alexeyklyukin) [inline helpers/postgresql.py]: We can, although it's the question of whether ON_START is less magical than "on_start" :-)

File: helpers/postgresql.py (hunk 1 above)
Comment (alexeyklyukin) [inline helpers/postgresql.py]: Good point.

**Pull request merged**